### PR TITLE
Added support of 32bit windows to socketmon

### PR DESCRIPTION
**Pull request by nystilaps**
Added support of 32bit windows to socketmon

### CODE REVIEW

**Comment by tklengyel** [comment]
@drakvuf-jenkins This is OK to test.

File: src/plugins/socketmon/socketmon.cpp [hunk 1]
```diff
@@ -1443,54 +1434,72 @@ static event_response_t trap_DnsQuery_W_cb(drakvuf_t drakvuf, drakvuf_trap_info_
...
-
+    vmi_free_unicode_str(domain_name_us);
 
     return 0;
 }
 
+// This is yet another type of Windows string representation
+// specific for undocumented DnsQueryExW(...) function.
+// Same type for 64 and 32 bit versions.
+struct dns_query_ex_w_string_t
```
Comment (tklengyel) [inline src/plugins/socketmon/socketmon.cpp]: Please move structure definitions into `private.h`

File: src/plugins/socketmon/socketmon.cpp [hunk 2]
```diff
@@ -1443,54 +1434,72 @@ static event_response_t trap_DnsQuery_W_cb(drakvuf_t drakvuf, drakvuf_trap_info_
...
     socketmon* sm = (socketmon*)info->trap->data;
 
-    if (sm->pm != VMI_PM_IA32E) // only 64 bit for now
+    addr_t domain_name_addr = drakvuf_get_function_argument(drakvuf, info, 1);
+
     {
-        PRINT_DEBUG("[SOCKETMON] Suddenly, we are in 32 bit mode in %s(...) trap. Unsupported.\n", info->trap->name);
-        return 0;
-    }
+        dns_query_ex_w_string_t strange_string;
```
Comment (tklengyel) [inline src/plugins/socketmon/socketmon.cpp]: This variable name begs for more information :)

Comment (nystilaps) [reply src/plugins/socketmon/socketmon.cpp]: Renamed into 'function_specific_string'

File: src/plugins/socketmon/socketmon.cpp (hunk 1 above)
Comment (nystilaps) [reply src/plugins/socketmon/socketmon.cpp]: Done

**Comment by nystilaps** [comment]
Made requested changes

**Comment by tklengyel** [comment]
@drakvuf-jenkins Retest this please

**Comment by tklengyel** [comment]
Thanks!

**Comment by nystilaps** [comment]
Made requested changes.

**Pull request merged**